tests/core/v5_1/test_v51_rpc_handlers.py

```diff
@@ -129,6 +134,36 @@ async def test_v51_rpc_ping_web3(make_request, bob_node_id_param_w3, alice, bob,
     assert pong.packet_port == alice.endpoint.port
 
 
+@pytest.mark.trio
+async def test_v51_rpc_send_ping(make_request, bob_client, bob):
```

Comment:
@pipermerriam When using the `bob_node_id_param` fixture here (as well as the `bob_node_id_param_w3` fixture for the `test_v51_rpc_send_ping_web3` test) it causes a strange error that I'm unfamiliar with: `trio.WouldBlock: Exception: Locked: owner=Node.network(...)  acquiring-for=Node.client(...)`. 

This error goes away when using `repr(bob.enr)` as the node id - but it seems to me that we would want to test all the node id flavors in these tests. Do you have an idea as to what's causing this failure?

Reply:
Ah, ok, so that's some safety code I wrote to ensure that you don't use `bob_network` and `bob_client` at the same time.  The reason is that `bob_network` runs the client under the hood so you'll run into an error trying to bind to the socket because the port will already be used.

So if you need both `bob_client` and `bob_network` within the same test, you should use `bob_network` and pull the client directly off of it as `bob_network.client`